src/config.rs

```diff
@@ -29,103 +28,119 @@ pub(crate) struct Config {
-
-        Self { directory, node: node.configuration() }
-    }
-
-    pub fn devnet(directory: PathBuf, node_name: String, is_executor: bool) -> Self {
-        let mut node = Node::devnet()
-            .network(NetworkBuilder::devnet().name(node_name))
-            .dsn(DsnBuilder::devnet().provider_storage_path(provider_storage_dir_getter()));
-
-        if is_executor {
-            node = node
-                .system_domain(domains::ConfigBuilder::new().core(ConfigBuilder::new().build()));
-        }
-        node = node.role(Role::Authority);
-
-        Self { directory, node: node.configuration() }
-    }
+/// Advanced Farmer Settings Wrapper for CLI
+#[derive(Deserialize, Serialize, Clone, Derivative, Default)]
+pub(crate) struct AdvancedFarmerSettings {
+    #[serde(with = "bytesize_serde", default, skip_serializing_if = "crate::utils::is_default")]
+    #[derivative(Default(value = "bytesize::ByteSize::gb(1)"))]
+    pub(crate) cache_size: ByteSize,
 }
 
-/// structure for the `farmer` field of the config toml file
-#[derive(Deserialize, Serialize)]
+/// Farmer Options Wrapper for CLI
+#[derive(Deserialize, Serialize, Clone)]
 pub(crate) struct FarmerConfig {
-    pub(crate) address: PublicKey,
+    pub(crate) reward_address: PublicKey,
     pub(crate) plot_directory: PathBuf,
     #[serde(with = "bytesize_serde")]
     pub(crate) plot_size: ByteSize,
-    #[serde(flatten)]
-    pub(crate) farmer: SdkFarmerConfig,
-    pub(crate) cache: CacheDescription,
+    pub(crate) advanced: AdvancedFarmerSettings,
 }
 
 impl FarmerConfig {
-    pub fn gemini_3c(
-        address: PublicKey,
-        plot_directory: PathBuf,
-        plot_size: ByteSize,
-        cache: CacheDescription,
-    ) -> Self {
-        Self {
-            address,
-            plot_directory,
-            plot_size,
-            cache,
-            farmer: Farmer::builder().configuration(),
+    pub async fn build(self, chain: ChainConfig, node: Node) -> Result<Farmer> {
+        let plot_description = &[PlotDescription::new(self.plot_directory, self.plot_size)
+            .wrap_err("Plot size is too low")?];
+        let cache = CacheDescription::new(cache_directory_getter(), self.advanced.cache_size)?;
+        // currently we do not have different configuration for the farmer w.r.t
+        // different chains, but we may in the future
+        match chain {
+            ChainConfig::Gemini3c => Farmer::builder()
+                .build(self.reward_address, node, plot_description, cache)
+                .await
+                .map_err(Report::msg),
+            ChainConfig::Dev => Farmer::builder()
+                .build(self.reward_address, node, plot_description, cache)
+                .await
+                .map_err(Report::msg),
+            ChainConfig::DevNet => Farmer::builder()
+                .build(self.reward_address, node, plot_description, cache)
+                .await
+                .map_err(Report::msg),
```

Comment:
it doesn't hurt the codebase imo, and I prefer it like as it is to remind myself. Gives the big-picture better :)

Reply:
I mean, it is not meaningful though. I can understand the idea from the comment, but `match` without the comment doesn't make sense. I think you can introduce that block, once `Farmer` configuration will diverge, from chain to chain, but I don't think it will ever happen.